Helper/Discount.php

```diff
@@ -979,6 +979,7 @@ public function applyMageplazaGiftCard($code, $quote)
             $giftCardsData = $this->sessionHelper->getCheckoutSession()->getGiftCardsData();
             $giftCardsData[self::MAGEPLAZA_GIFTCARD_QUOTE_KEY][$code] = 0;
             $this->sessionHelper->getCheckoutSession()->setGiftCardsData($giftCardsData);
+            $quote->setMpGiftCards(json_encode([$code => 0]));
```

Comment:
@johnmccombs1 Could you please explain the fix?

Reply:
@vitaliyreznikov  the reason I added this change is that on the older MagePlaza version, they used the session to save the gift card codes and then pull them out to collect the total. However, in the latest version, they save the gift card to the quote to collect the total.

FYI, [$code => 0] might a bit confusing but it is how they use it. See here - https://prnt.sc/tvpkv9

Reply:
Oh, I see.
So we support both versions by this code?
If so, could you please add a comment that line 981 for versions below ... and line 982 for versions above ...

Reply:
@vitaliyreznikov I've added some comments. Can you re-check?

Reply:
Could you please add the specific version number? It's the lastest for now, but they can update algorithm again later.

Reply:
I added the specific version number. Can you re-check?